### DISCUSSION TITLE
InputNumber does not support a size attribute

### DISCUSSION THREAD

**Discussion by amurf** [primefaces/community#2493] [PrimeVue]
As per https://github.com/primefaces/primevue/issues/4969

InputText supports a size attribute (https://primevue.org/inputtext/#sizes), but none of the other input fields do, which i feels unexpected/inconsistent.

For now I'm working around this by using  `input-class="p-inputtext-sm"` on the InputNumber, but I feel like accepting size on the input components makes sense and is cleaner.

**Comment by BenJackGill** [comment]
I agree.

Also the `Password` component has the same problem, no size prop.

Ideally all "input" type components should have a size prop. This is needed so forms with a mix of inputs (`InputText`, `Password`, `InputNumber`, etc) can all have small sized inputs.

**Comment by BenJackGill** [comment]
There are los of discussions around this but no traction on it yet. In an attempt to get more visibility to the problem here are some of the other discussions asking for a `size` prop on all form components:

- https://github.com/orgs/primefaces/discussions/2493
- https://github.com/orgs/primefaces/discussions/2684
- https://github.com/orgs/primefaces/discussions/2381
- https://github.com/orgs/primefaces/discussions/2809

I love PrimeVue! So hoping they can add this soon.
